proguard6.0.3/core/src/proguard/optimize/info/ParameterEscapedMarker.java

```diff
@@ -0,0 +1,307 @@
+
+    // Parameters and values for visitor methods.
+    private boolean newEscapes;
+    private Method  referencingMethod;
+    private int     referencingOffset;
+    private int     referencingPopCount;
+
+
+    /**
+     * Creates a new ParameterModificationMarker.
+     */
+    public ParameterEscapedMarker()
+    {
+    }
+
+
+    // Implementations for ClassPoolVisitor.
+
+    public void visitClassPool(ClassPool classPool)
+    {
+        // Go over all classes and their methods, marking if parameters are
+        // modified, until no new cases can be found.
+        do
+        {
+            newEscapes = false;
+
+            if (DEBUG)
+            {
+                System.out.println("ParameterEscapedMarker: new iteration");
+            }
+
+            // Go over all classes and their methods once.
+            classPool.classesAccept(parameterEscapedMarker);
+        }
+        while (newEscapes);
+
+        if (DEBUG)
+        {
+            classPool.classesAccept(new AllMethodVisitor(this));
+        }
+    }
+
+
+    // Implementations for MemberVisitor.
+
+    public void visitProgramMethod(ProgramClass programClass, ProgramMethod programMethod)
+    {
+        if (DEBUG)
+        {
+            System.out.println("ParameterEscapedMarker: [" + programClass.getName() + "." + programMethod.getName(programClass) + programMethod.getDescriptor(programClass) + "]");
+
+            int parameterSize =
+                ClassUtil.internalMethodParameterSize(programMethod.getDescriptor(programClass),
+                                                      programMethod.getAccessFlags());
+
+            for (int index = 0; index < parameterSize; index++)
+            {
+                System.out.println("  " +
+                                   (hasParameterEscaped(programMethod, index) ? 'e' : '.') +
+                                   " P" + index);
+            }
+        }
+    }
+
+
+    // Implementations for AttributeVisitor.
+
+    public void visitAnyAttribute(Clazz clazz, Attribute attribute) {}
+
+
+    public void visitCodeAttribute(Clazz clazz, Method method, CodeAttribute codeAttribute)
+    {
+        // Evaluate the code.
+        partialEvaluator.visitCodeAttribute(clazz, method, codeAttribute);
+        referenceEscapeChecker.visitCodeAttribute(clazz, method, codeAttribute);
+
+        // Mark the parameters that are modified from the code.
+        codeAttribute.instructionsAccept(clazz, method, partialEvaluator.tracedInstructionFilter(this));
+    }
+
+
+    // Implementations for InstructionVisitor.
+
+    public void visitAnyInstruction(Clazz clazz, Method method, CodeAttribute codeAttribute, int offset, Instruction instruction) {}
+
+
+    public void visitConstantInstruction(Clazz clazz, Method method, CodeAttribute codeAttribute, int offset, ConstantInstruction constantInstruction)
+    {
+        switch (constantInstruction.opcode)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Switch statements should have a default label](https://app.codacy.com/manual/airsquared/blobsaver/pullRequest?prid=5513820)